frontend/src/v5/store/projects/projects.sagas.ts

```diff
@@ -70,9 +72,40 @@ export function* deleteProject({ teamspace, projectId, onSuccess, onError }) {
 	}
 }
 
+export function* fetchTemplates({ teamspace, projectId }) {
+	try {
+		let isFed = false;
+		let model = (yield select(selectContainers))?.[0];
+		if (!model) {
+			isFed = true;
+			model = (yield select(selectFederations))?.[0];
+		}
+		if (!model) {
+			throw new Error(
+				formatMessage({
+					id: 'projects.error.noModels',
+					defaultMessage: 'The project must contain at least one container or federation to fetch the templates',
+				}),
+			);
+		}
+		const fetchModelTemplates = isFed ? API.Tickets.fetchFederationTemplates : API.Tickets.fetchContainerTemplates;
```

Comment:
At the moment, teamspace endpoint would indeed return the same templates as any model, but this is not guaranteed to be the case as it may change in the future. Ideally, we're going to have a project endpoints for templates in the future, but for now, this is the way